src/main/java/org/junit/Assert.java

```diff
@@ -991,6 +996,8 @@ public static void assertThrows(Class<? extends Throwable> expectedThrowable, Th
     public static <T extends Throwable> T expectThrows(Class<T> expectedThrowable, ThrowingRunnable runnable) {
         try {
             runnable.run();
+        } catch (AssumptionViolatedException e) {
+            throw new AssumptionNotSupportedException("Assumptions are not supported in expectThrows/assertThrows", e);
```

Comment:
also, I don't see how this helps, since befoer your change if `AssumptionViolatedException` was not expected and was thrown, it would be wrapped with an `AssertionError`